### PR TITLE
[#5625] moving upload and uploadFromFile to BlobClient

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes https://github.com/Azure/azure-sdk-for-java/issues/5625

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/SpecializedBlobClientBuilder.java [hunk 1]
```diff
@@ -86,7 +90,8 @@ public BlockBlobClient buildBlockBlobClient() {
 
     /**
      * Creates a {@link BlockBlobAsyncClient} based on options set in the Builder. BlockBlobAsyncClients are used to
-     * perform generic upload operations such as {@link BlockBlobAsyncClient#uploadFromFile(String) upload from file}
+     * perform generic upload operations such as {@link BlockBlobAsyncClient#uploadFromFile(String, Integer,
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/SpecializedBlobClientBuilder.java]: This method was removed from the block blob client, instead reference it's upload method,

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobAsyncClient.java [hunk 2]
```diff
@@ -163,48 +163,6 @@
...
-     * @param blockSize The size of each block that will be staged. This value also determines the size that each buffer
-     * used by this method will be and determines the number of requests that need to be made. The amount of memory
-     * consumed by this method may be up to blockSize * numBuffers. If block size is large, this method will make fewer
-     * network calls, but each individual call will send more data and will therefore take longer.
-     * @param numBuffers The maximum number of buffers this method should allocate. Must be at least two. Typically, the
-     * larger the number of buffers, the more parallel, and thus faster, the upload portion of this operation will be.
-     * The amount of memory consumed by this method may be up to blockSize * numBuffers.
-     * @return A reactive response containing the information of the uploaded block blob.
-     */
-    public Mono<BlockBlobItem> upload(Flux<ByteBuffer> data, int blockSize, int numBuffers) {
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobAsyncClient.java]: Move all the upload methods like this and uploadFromFile methods to BlobClient. Move them for the sync client as well.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java [hunk 3]
```diff
@@ -96,4 +132,245 @@ private SpecializedBlobClientBuilder prepareBuilder() {
 
         return builder;
     }
+
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java]: Should the regular upload (the one that takes the data and length?) also be ported over to this class?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java [hunk 4]
```diff
@@ -85,4 +98,48 @@ public PageBlobClient getPageBlobClient() {
...
+     * Creates a new block blob, or updates the content of an existing block blob.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobClient.uploadFromFile#String}
+     *
+     * @param filePath Path of the file to upload
+     * @throws IOException If an I/O error occurs
+     */
+    public void uploadFromFile(String filePath) throws IOException {
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java]: For this should upload(InputStream, long) be also ported over here?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java (hunk 3 above)
Comment (vhvb1989) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java]: think we are following:
```
In a broader spectrum the BlobClient is meant to contain all methods that aren't 1 to 1 with the REST specification where the specialized (Append, Block, and Page clients) are 1 to 1 with the REST specification.
```

and upload(data, length) is 1 to 1 to `put` blob with a length header?  in that case it is expected to be in specialized package

**Pull request merged**